sherpa/ui/utils.py

```diff
@@ -973,6 +973,53 @@ def clean(self):
             'source_component': sherpa.image.ComponentSourceImage()
         }
 
+        # Reset the generator.
+        #
+        self.set_rng(None)
+
+    def get_rng(self):
+        """Return the RNG generator in use.
+
+        The return can be None, which means the legacy NumPy
+        RandomState generator is in use: see
+        https://numpy.org/doc/stable/reference/random/legacy.html
+
+        .. versionadded:: 4.16.0
+
+        See Also
+        --------
+        set_rng
+
+        """
+
+        return self._rng
+
+    def set_rng(self, rng):
+        """Set the RNG generator.
+
+        .. versionadded:: 4.16.0
+           This replaces the seed argument for certain routines and
+           the need to call np.random.seed directly in others.
+
+        Parameters
+        ----------
+        rng : numpy.random.Generator or None
```

Comment:
We could require that `rng` be a generator, but this would make it hard for users to match the code to previous versions which used `np.random.seed` to set a "known" random state. The answer could be `you need to change np.random.seed(xxx) to ui.set_rng(np.random.RandomState(xxx))` but this is not "user-friendly".

Reply:
Why is that not user-friendly? Because it's not automatic?

Reply:
Because it requires users to change code that used to work.